src/contrib/persistence/Akka.Persistence.Sql.Common/Journal/BatchingSqlJournal.cs

```diff
@@ -837,7 +875,8 @@ private void TryProcess()
 
                 var chunk = DequeueChunk(_remainingOperations);
                 var context = Context;
-                _circuitBreaker.WithCircuitBreaker(() => ExecuteChunk(chunk, context)).PipeTo(Self);
+                _circuitBreaker.WithCircuitBreaker(() => ExecuteChunk(chunk, context))
```

Comment:
Halfway there, but still when the circuit breaker throws an `OpenCircuitException` the restarts won't happen again. Wrapping this piece in a try...catch... and self issuing a `new ChunkExecutionFailure(ex, chunk.Requests)` seems to do the trick.

Reply:
Are you saying that `OpenCircuitException` is not propagated to `PipeTo`?

Reply:
The `PipeTo` will work

Reply:
Unless @ismaelhamed means that the `_circuitBreaker` itself throws instead of injecting the exception into its `Task`, in which case his feedback is correct.

Reply:
Seems like the `_circuitbreaker` is throwing instead of propagating to `PipeTo`, hence why I had to wrap in a try...catch... that call to `_circuitBreaker.WithCircuitBreaker()` in order for it to work. 

I´ll do some more tests tomorrow to confirm it.

Reply:
> Seems like the _circuitbreaker is throwing instead of propagating to PipeTo, hence why I had to wrap in a try...catch... that call to _circuitBreaker.WithCircuitBreaker() in order for it to work.

Not to get side-tracked, but if that's the case then isn't that a bug with the way the `CircuitBreaker` is designed?

Reply:
I've updated `CircuitBreaker` implementation to raise exceptions withing returned `Task`s (simply by marking corresponding methods as async).